components/micro-gateway-core/src/main/java/org/wso2/micro/gateway/core/callhome/Callhome.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.micro.gateway.core.callhome;
+
+import org.wso2.callhome.CallHomeExecutor;
+import org.wso2.callhome.utils.CallHomeInfo;
+import org.wso2.callhome.utils.Util;
+import org.wso2.micro.gateway.core.Constants;
+
+import java.io.File;
+/**
+ * Invoke call home
+ *
+ *
+ */
+public class Callhome {
+
+    /**
+     * run call home
+     *
+     */
+    public static void runCallHome() {
+        String productHome = getRuntimeHome();
+        String trustStoreLocation = getTrustStoreLocation();
+        String trustStorePassword = Constants.TRUST_STORE_PASSWORD;
+
+        CallHomeInfo callhomeinfo = Util.createCallHomeInfo(productHome, trustStoreLocation, trustStorePassword);
+        CallHomeExecutor.execute(callhomeinfo);
+    }
+
+    /**
+     * Get runtime home location
+     *
+     * @return runtime home location
+     */
+    private static String getRuntimeHome() {
+        return System.getProperty(Constants.RUNTIME_HOME_PATH);
+    }
+
+    /**
+     * Get truststore location
+     *
+     * @return truststore location
+     */
+    private static String getTrustStoreLocation() {
+        return getRuntimeHome() + File.separator + Constants.TRUST_STORE_LOCATION;
```

Comment:
Truststore location is a configurable parameter in the gateway. If it is changed, will this work?